lib/on-contract-event.js

```diff
@@ -0,0 +1,100 @@
+// eth_getFilterChanges isn't fully supported by lotus, see
+// https://filecoinproject.slack.com/archives/CRK2LKYHW/p1705404161018769
+// Therefore we need to poll for events
+
+import { ethers } from 'ethers'
+import timers from 'node:timers/promises'
+
+const MIN_POLL_INTERVAL = 10_000
+
+const makeRequest = async ({
+  contract,
+  address,
+  rpcUrl,
+  rpcHeaders,
+  lastBlockNumber
+}) => {
+  const req = {
+    jsonrpc: '2.0',
+    id: 1,
```

Comment:
Iiuc id is only unique in order to match responses to requests, in long running connections. Since we're opening new connections on every loop, this isn't a problem

Reply:
> we're opening new connections on every loop, this isn't a problem

Is it worth capturing this information in a code comment?

Reply:
I don't think so, if we were parsing multiple responses from the rpc endpoint and matching the `id` then it would be relevant, but imo codebases don't need to be knowledgebases, unless it becomes a recurring situation. Happy to add the comment though if you still think it's benefitial